hydromt/models/model_mesh.py

```diff
@@ -304,17 +474,76 @@ def write_mesh(self, fn: str = "mesh/mesh.nc", **kwargs) -> None:
             os.makedirs(dirname(_fn))
         self.logger.debug(f"Writing file {fn}")
         ds_out = self.mesh.ugrid.to_dataset()
-        if self.mesh.ugrid.grid.crs is not None:
+        if self.crs is not None:
             # save crs to spatial_ref coordinate
-            ds_out = ds_out.rio.write_crs(self.mesh.ugrid.grid.crs)
+            ds_out = ds_out.rio.write_crs(self.crs)
         ds_out.to_netcdf(_fn, **kwargs)
 
+    # Other mesh properties
+    @property
+    def mesh_grids(self) -> Dict:
+        """Dictionnary of grid names and Ugrid topologies in mesh."""
+        grids = dict()
+        if self.mesh is not None:
+            for grid in self.mesh.ugrid.grids:
+                grids[grid.name] = grid
+
+        return grids
+
+    @property
+    def mesh_datasets(self) -> Dict:
+        """Dictionnary of grid names and corresponding UgridDataset topology and data variables in mesh."""  # noqa: E501
+        datasets = dict()
+        if self._mesh is not None:
+            for grid in self.mesh.ugrid.grids:
+                datasets[grid.name] = self.get_mesh(
+                    grid_name=grid.name, include_data=True
+                )
+
+        return datasets
+
+    @property
+    def mesh_names(self) -> List[str]:
+        """List of grid names in mesh."""
+        # FIXME would be nice if xugrid supports it: https://github.com/Deltares/xugrid/issues/141
+        if self.mesh is not None:
+            return list(self.mesh_grids.keys())
+        else:
+            return []
+
+    @property
+    def mesh_gdf(self) -> Dict:
+        """Returns dict of geometry of grids in mesh as a gpd.GeoDataFrame."""
+        mesh_gdf = dict()
+        if self._mesh is not None:
+            for k, v in self.mesh_datasets.items():
+                # works better on a Dataarray / Dataset
+                # (need to add dummy variable if empty)
+                if len(v.data_vars) == 0:
+                    grid = v.ugrid.grid
+                    if grid.topology_dimension == 1:
+                        dim = grid.edge_dimension
+                    elif grid.topology_dimension == 2:
+                        dim = grid.face_dimension
+                    v[f"{grid.name}_node_id"] = xr.DataArray(
+                        v[dim].values.astype(str), dims=dim
+                    )
+                gdf = v.ugrid.to_geodataframe()
+                # FIXME xugrid crs not passed on correctly: https://github.com/Deltares/xugrid/issues/138
+                mesh_gdf[k] = gdf.set_crs(v.ugrid.crs[k])
+
+        return mesh_gdf
```

Comment:
Hi @hboisgon, when testing the building and updating of mesh2d, I found that the `to_geodataframe()` function does not always work if 1) there is no data variable (I guess thas why you first assigned a new node_id data variable) 2) there are optiaonl attributes besides face coordinates. Since this is just to return the feometry of the mesh, would it suffice if we skip 'to_geodataframe()' func and create a simple one?
```suggestion
    def mesh_gdf(self) -> Dict:
        """Returns dict of geometry of grids in mesh as a gpd.GeoDataFrame."""
        mesh_gdf = dict()
        if self._mesh is not None:
            for k, v in self.mesh_grids.items():
                # works better on a Dataarray / Dataset
                # (need to add dummy variable if empty)
                if len(v.data_vars) == 0:
                    grid = v.ugrid.grid
                    if grid.topology_dimension == 1:
                        dim = grid.edge_dimension
                    elif grid.topology_dimension == 2:
                        dim = grid.face_dimension
                gdf = gpd.GeoDataFrame(
                    index=v[dim].values.astype(str),
                    geometry=grid.to_shapely(dim),
                )
                mesh_gdf[k] = gdf.set_crs(v.ugrid.crs[k])

        return mesh_gdf
```

Reply:
Sounds good for now and avoid the error yes. Maybe then if it's useful we could add a flag to also add data variables to it, but let's put that on the wish list. Maybe still good to report these bugs to xugrid though as they may want to look into this as well.